### PR TITLE
Apply backend naming convention

### PR DESCRIPTION
**Pull request by mostafa**
Related to #110: https://github.com/SigmaHQ/pySigma/discussions/110#discussioncomment-6179482

### CODE REVIEW

**Comment by thomaspatzke** [comment]
Argh, sorry, I've made a mistake...forgot about it in our discussion that *name* already exists as descriptive name [like here](https://github.com/SigmaHQ/pySigma-backend-splunk/blob/bfcbbefa6a77c481fedbed63f2860ff6a1e153de/sigma/backends/splunk/splunk.py#L31). These usually contain spaces and are long, what is painful for CLI usage.

I think it's easier for backwards compatibility to use another class variable name *identifier* for this purpose. Can you change this? 

Can you further add a test for the backend name generation?

Anyways...great contribution, I've already removed it from my todo list 😁

File: tests/test_backend_identifier.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    pass
+
+
+class BaseBackend(TextQueryBackend):
+    """Dummy backend for testing purposes."""
+
+    pass
+
+
+@pytest.mark.parametrize(
```
Comment (mostafa) [inline tests/test_backend_identifier.py:57]: I hope I added enough test cases and it actually addresses your comment.

File: sigma/plugins.py [hunk 2]
```diff
@@ -180,6 +206,62 @@ def get_pipeline_resolver(self) -> ProcessingPipelineResolver:
...
+            The identifier of the backend object in snake_case or the default identifier.
+        """
+
+        def removesuffix(base: str, suffix: str) -> str:
+            """Removes the suffix from the string if it exists.
+            This is a backport of the Python 3.9 removesuffix method.
+            """
+            if base.endswith(suffix):
+                return base[: len(base) - len(suffix)]
+            return base
```
Comment (mostafa) [inline sigma/plugins.py:231]: This is introduced in Python 3.9, which is why tests failed on earlier commits when running on Python 3.8.

**Comment by thomaspatzke** [comment]
Great, thanks a lot!

**Pull request merged**